### PR TITLE
releases: add tidb 3.0.20 release notes

### PR DESCRIPTION
**Pull request by ti-srebot**
### What is changed, added or deleted? (Required)

add tidb 3.0.20 release notes

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):
- docs-cn notes: https://github.com/pingcap/docs-cn/pull/5106

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: releases/release-3.0.20.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    - Deprecate `enable-streaming`. [#21054](https://github.com/pingcap/tidb/pull/21054)
+    - Fix compatibility for `ADDDATE` function [#21008](https://github.com/pingcap/tidb/pull/21008)
+    - Set correct default value for new added enum column. [#20999](https://github.com/pingcap/tidb/pull/20999)
+    - Fix the result of SQL `select DATE_ADD('2007-03-28 22:08:28',INTERVAL "-2.-2" SECOND)` to be same with the MySQL; [#20627](https://github.com/pingcap/tidb/pull/20627)
+    - Fix #20490, incorrect default value when modify column [#20532](https://github.com/pingcap/tidb/pull/20532)
+    - Fix function timestamp() get wrong result when input argument is type of float/decimal [#20469](https://github.com/pingcap/tidb/pull/20469)
+    - Statistics: fix an unsafe lock operation and adjust some logging levels [#20424](https://github.com/pingcap/tidb/pull/20424)
+    - Fix FLOAT data type: out of range data should not be inserted [#20251](https://github.com/pingcap/tidb/pull/20251)
+
++ TiKV
```
Comment (youjiali1995) [inline releases/release-3.0.20.md:44]: lgtm

File: releases/release-3.0.20.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    - Add end_point_slow_log_threshold config [#9145](https://github.com/tikv/tikv/pull/9145)
```
Comment (youjiali1995) [inline releases/release-3.0.20.md:46]: Improvements

File: releases/release-3.0.20.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    - Fix the issue that reports a key-exist error when a key is locked and deleted by a committed transaction. [#8931](https://github.com/tikv/tikv/pull/8931)
```
Comment (youjiali1995) [inline releases/release-3.0.20.md:47]: Bug Fixes

File: releases/release-3.0.20.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    - Fix the statistics are inaccurate when querying `INFORMATION_SCHEMA.TIDB_HOT_REGIONS` [#21319](https://github.com/pingcap/tidb/pull/21319)
+    - Raise an error when preparing the `load data` statement. [#21222](https://github.com/pingcap/tidb/pull/21222)
+    - `DELETE` may not delete data correctly when the database name is not in pure lower representation. [#21205](https://github.com/pingcap/tidb/pull/21205)
+    - Fix stack overflow when building recursive view. [#21000](https://github.com/pingcap/tidb/pull/21000)
+    - Fix an goroutine leak issue in TiKV client. [#20863](https://github.com/pingcap/tidb/pull/20863)
+    - Fix wrong default zero value for year type [#20828](https://github.com/pingcap/tidb/pull/20828)
+    - Avoid goroutine leak in index lookup join [#20791](https://github.com/pingcap/tidb/pull/20791)
+    - Fix "insert select for update" return malformed packet in pessimistic txn [#20681](https://github.com/pingcap/tidb/pull/20681)
+    - Fix unknown time zone 'posixrules'. [#20605](https://github.com/pingcap/tidb/pull/20605)
+    - Fix data too long when converting to bit returns a null value [#20362](https://github.com/pingcap/tidb/pull/20362)
```
Comment (SunRunAway) [inline releases/release-3.0.20.md:25]:
```suggestion
    - Fix a problem of data too long when converting from int unsigned it to bit [#20362](https://github.com/pingcap/tidb/pull/20362)
```

File: releases/release-3.0.20.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    - Fix corrupted default value for bit type column. [#20339](https://github.com/pingcap/tidb/pull/20339)
+    - Avoid propagate column optimization for Hybrid type. [#20296](https://github.com/pingcap/tidb/pull/20296)
```
Comment (SunRunAway) [inline releases/release-3.0.20.md:27]:
```suggestion
    - Fix potentially incorrect results when one of the equal condition is Enum of Set type. [#20296](https://github.com/pingcap/tidb/pull/20296)
```

File: releases/release-3.0.20.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    - Fix wrong behavior for != any() [#20061](https://github.com/pingcap/tidb/pull/20061)
+
+## Others
+
++ TiDB
+
+    - Fix an issue that invalid results using BETWEEN...AND... with type conversion [#21503](https://github.com/pingcap/tidb/pull/21503)
```
Comment (SunRunAway) [inline releases/release-3.0.20.md:34]: This line should be moved to bug fix.

File: releases/release-3.0.20.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    - Deprecate `enable-streaming`. [#21054](https://github.com/pingcap/tidb/pull/21054)
+    - Fix compatibility for `ADDDATE` function [#21008](https://github.com/pingcap/tidb/pull/21008)
+    - Set correct default value for new added enum column. [#20999](https://github.com/pingcap/tidb/pull/20999)
```
Comment (SunRunAway) [inline releases/release-3.0.20.md:37]: This line should be moved to bug fix.

File: releases/release-3.0.20.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    - Fix the result of SQL `select DATE_ADD('2007-03-28 22:08:28',INTERVAL "-2.-2" SECOND)` to be same with the MySQL; [#20627](https://github.com/pingcap/tidb/pull/20627)
+    - Fix #20490, incorrect default value when modify column [#20532](https://github.com/pingcap/tidb/pull/20532)
+    - Fix function timestamp() get wrong result when input argument is type of float/decimal [#20469](https://github.com/pingcap/tidb/pull/20469)
+    - Statistics: fix an unsafe lock operation and adjust some logging levels [#20424](https://github.com/pingcap/tidb/pull/20424)
+    - Fix FLOAT data type: out of range data should not be inserted [#20251](https://github.com/pingcap/tidb/pull/20251)
```
Comment (SunRunAway) [inline releases/release-3.0.20.md:42]: These lines should be moved to bug fix.

File: releases/release-3.0.20.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    - Avoid propagate column optimization for Hybrid type. [#20296](https://github.com/pingcap/tidb/pull/20296)
+    - Fix wrong behavior for != any() [#20061](https://github.com/pingcap/tidb/pull/20061)
+
+## Others
+
++ TiDB
+
+    - Fix an issue that invalid results using BETWEEN...AND... with type conversion [#21503](https://github.com/pingcap/tidb/pull/21503)
+    - Deprecate `enable-streaming`. [#21054](https://github.com/pingcap/tidb/pull/21054)
+    - Fix compatibility for `ADDDATE` function [#21008](https://github.com/pingcap/tidb/pull/21008)
```
Comment (SunRunAway) [inline releases/release-3.0.20.md:36]: This line should be moved to bug fix.

File: releases/release-3.0.20.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+    - Set correct default value for new added enum column. [#20999](https://github.com/pingcap/tidb/pull/20999)
+    - Fix the result of SQL `select DATE_ADD('2007-03-28 22:08:28',INTERVAL "-2.-2" SECOND)` to be same with the MySQL; [#20627](https://github.com/pingcap/tidb/pull/20627)
+    - Fix #20490, incorrect default value when modify column [#20532](https://github.com/pingcap/tidb/pull/20532)
```
Comment (SunRunAway) [inline releases/release-3.0.20.md:39]:
```suggestion
    - Fix incorrect default value when modify column [#20532](https://github.com/pingcap/tidb/pull/20532)
```

File: releases/release-3.0.20.md [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+    - Fix function timestamp() get wrong result when input argument is type of float/decimal [#20469](https://github.com/pingcap/tidb/pull/20469)
+    - Statistics: fix an unsafe lock operation and adjust some logging levels [#20424](https://github.com/pingcap/tidb/pull/20424)
```
Comment (SunRunAway) [inline releases/release-3.0.20.md:41]:
```suggestion
    - fix a potentially deadlock problem in statistic and adjust some logging levels [#20424](https://github.com/pingcap/tidb/pull/20424)
```

File: releases/release-3.0.20.md [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+    - Fix FLOAT data type: out of range data should not be inserted [#20251](https://github.com/pingcap/tidb/pull/20251)
+
++ TiKV
+
+    - Add end_point_slow_log_threshold config [#9145](https://github.com/tikv/tikv/pull/9145)
+    - Fix the issue that reports a key-exist error when a key is locked and deleted by a committed transaction. [#8931](https://github.com/tikv/tikv/pull/8931)
+
++ PD
```
Comment (Yisaer) [inline releases/release-3.0.20.md:49]: LGTM

**Comment by TomShawn** [comment]
@SunRunAway @youjiali1995 PTAL again, thanks~

**Review by youjiali1995 — APPROVED** [review]
lgtm

**Review by SunRunAway — COMMENTED** [review]
LGTM

File: releases/release-3.0.20.md [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
++ TiKV
+
+    - Add the `end_point_slow_log_threshold` configuration item [#9145](https://github.com/tikv/tikv/pull/9145)
+
+## Bug Fixes
+
++ TiDB
+
+    - Fix the resolved txn status cache for pessimistic transactions. [#21706](https://github.com/pingcap/tidb/pull/21706)
+    - Fix the statistics are inaccurate when querying `INFORMATION_SCHEMA.TIDB_HOT_REGIONS` [#21319](https://github.com/pingcap/tidb/pull/21319)
```
Comment (TomShawn) [inline releases/release-3.0.20.md:32]:
```suggestion
    - Fix the issue of inaccurate statistics that occurs when querying `INFORMATION_SCHEMA.TIDB_HOT_REGIONS` [#21319](https://github.com/pingcap/tidb/pull/21319)
```

File: releases/release-3.0.20.md [hunk 14]
```diff
@@ -0,0 +1,60 @@
+---
+title: TiDB 3.0.20 Release Notes
+---
+
+# TiDB 3.0.20 Release Notes
+
+Release date: December 23, 2020
```
Comment (TomShawn) [inline releases/release-3.0.20.md:7]:
```suggestion
Release date: December 25, 2020
```

File: releases/release-3.0.20.md [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+
++ TiKV
+
+    - Add the `end_point_slow_log_threshold` configuration item [#9145](https://github.com/tikv/tikv/pull/9145)
+
+## Bug Fixes
+
++ TiDB
+
+    - Fix the resolved txn status cache for pessimistic transactions. [#21706](https://github.com/pingcap/tidb/pull/21706)
```
Comment (TomShawn) [inline releases/release-3.0.20.md:31]:
```suggestion
    - Fix the incorrect cache of the transaction status for pessimistic transactions [#21706](https://github.com/pingcap/tidb/pull/21706)
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**